packages/config-ui/src/settings/panel.jsx

```diff
@@ -275,20 +275,25 @@ export class Panel extends React.Component {
     const { groups, model, configuration } = this.props;
     log('render:', model);
 
-    return (
-      <div>
-        {Object.keys(groups).map((g) => (
-          <Group
-            label={g}
-            key={g}
-            model={model}
-            configuration={configuration}
-            group={groups[g]}
-            onChange={this.change}
-          />
-        ))}
-      </div>
-    );
+    const renderedGroups = Object.keys(groups)
+      .map((group) => {
+        const showGroup = Object.entries(groups[group]).some(([propName, propVal]) => !!propVal);
+        if (showGroup) {
+          return (
+            <Group
+              label={group}
+              key={group}
+              model={model}
+              configuration={configuration}
+              group={groups[group]}
+              onChange={this.change}
+            />
+          );
+        }
+      })
+      .filter((item) => !!item);
```

Comment:
Instead of filtering items, you can return null in the .map cycle if your condition was not fulfilled